impl/src/main/java/org/jboss/weld/bootstrap/events/BuilderInterceptorBean.java

```
@@ -0,0 +1,173 @@
+        types.add(Object.class);
+        return types;
+    }
+
+    @Override
+    public Set<Annotation> getQualifiers() {
+        return Collections.emptySet();
+    }
+
+    @Override
+    public Class<? extends Annotation> getScope() {
+        return Dependent.class;
+    }
+
+    @Override
+    public String getName() {
+        return getBeanClass().toString() + interceptionType.name() + priority + Formats.formatAnnotations(bindings);
+    }
+
+    @Override
+    public Set<Class<? extends Annotation>> getStereotypes() {
+        return Collections.emptySet();
+    }
+
+    @Override
+    public boolean isAlternative() {
+        return false;
+    }
+
+    protected WeldCreationalContext<?> getParentCreationalContext(CreationalContext<BuilderInterceptorInstance> ctx) {
+        if (ctx instanceof WeldCreationalContext<?>) {
+            WeldCreationalContext<?> parentContext = ((WeldCreationalContext<?>) ctx).getParentCreationalContext();
+            if (parentContext != null) {
+                return parentContext;
+            }
+        }
+        throw BeanLogger.LOG.unableToDetermineParentCreationalContext(ctx);
+    }
+
+    @Override
+    public int getPriority() {
+        return priority;
+    }
+
+    @Override
+    public BuilderInterceptorInstance create(CreationalContext<BuilderInterceptorInstance> creationalContext) {
+        BuilderInterceptorInstance interceptorInstance;
+
+        if (this.interceptorMetadataFunction != null) {
+            WeldCreationalContext<?> interceptorContext = getParentCreationalContext(creationalContext);
+            Contextual<?> interceptedContextual = interceptorContext.getContextual();
+            if (interceptedContextual instanceof Bean<?>) {
+                Bean<?> bean = (Bean<?>) interceptedContextual;
+                interceptorInstance = new BuilderInterceptorInstance(bean);
+                return interceptorInstance;
+            } else {
+                return null;
```

I think we should throw an exception here... WDYT?